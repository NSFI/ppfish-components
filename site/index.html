<!DOCTYPE html>
<html>
  <head>
    <title>Fish Design</title>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="shortcut icon" href="site/assets/favicon.ico">
    <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.css" />
  </head>
  <body>
    <div id="app"></div>
  </body>
</html>
