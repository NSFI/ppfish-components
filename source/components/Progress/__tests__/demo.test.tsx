import {plainComponents} from '../../../../site/componentsPage';

let demo_component = 'Progress';

let demoName = demo_component.substr(0, 1).toLowerCase() + demo_component.substr(1);
const menuItem = plainComponents.find(item => item.key === demoName);

if(menuItem){
  const demoTest = require('../../../../tools/tests/demoTest').default;
  demoTest(demoName);
}else{
  xit('skip',()=>{});
}
