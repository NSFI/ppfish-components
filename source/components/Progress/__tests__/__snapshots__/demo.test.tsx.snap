// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders progress demo correctly 1`] = `
<div>
  <div>
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:30%;height:8px;border-radius:100px"
              />
            </div>
          </div>
          <span
            class="fishd-progress-text"
          >
            30%
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-active fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:50%;height:8px;border-radius:100px"
              />
            </div>
          </div>
          <span
            class="fishd-progress-text"
          >
            50%
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-exception fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:70%;height:8px;border-radius:100px"
              />
            </div>
          </div>
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-error"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-success fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:100%;height:8px;border-radius:100px"
              />
            </div>
          </div>
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-success"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-normal fishd-progress-default"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:50%;height:8px;border-radius:100px"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style="width:170px"
  >
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-normal fishd-progress-show-info fishd-progress-small"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:30%;height:6px;border-radius:100px"
              />
            </div>
          </div>
          <span
            class="fishd-progress-text"
          >
            30%
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-active fishd-progress-show-info fishd-progress-small"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:50%;height:6px;border-radius:100px"
              />
            </div>
          </div>
          <span
            class="fishd-progress-text"
          >
            50%
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-exception fishd-progress-show-info fishd-progress-small"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:70%;height:6px;border-radius:100px"
              />
            </div>
          </div>
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-error"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-success fishd-progress-show-info fishd-progress-small"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:100%;height:6px;border-radius:100px"
              />
            </div>
          </div>
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-success"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="fishd-progress fishd-progress-circle fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-inner"
        style="width:120px;height:120px;font-size:24px"
      >
        <svg
          class="fishd-progress-circle "
          viewBox="0 0 100 100"
        >
          <path
            class="fishd-progress-circle-trail"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#f3f3f3"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
          />
          <path
            class="fishd-progress-circle-path"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#108ee9"
            stroke-linecap="round"
            stroke-width="6"
            style="stroke-dasharray:221.48228207808043px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
          />
        </svg>
        <div
          class="fishd-progress-circle-info"
        >
          <span
            class="fishd-progress-text"
          >
            75%
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-circle fishd-progress-status-exception fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-inner"
        style="width:120px;height:120px;font-size:24px"
      >
        <svg
          class="fishd-progress-circle "
          viewBox="0 0 100 100"
        >
          <path
            class="fishd-progress-circle-trail"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#f3f3f3"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
          />
          <path
            class="fishd-progress-circle-path"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#ff5500"
            stroke-linecap="round"
            stroke-width="6"
            style="stroke-dasharray:206.7167966062084px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
          />
        </svg>
        <div
          class="fishd-progress-circle-info"
        >
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-alone-error"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-circle fishd-progress-status-success fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-inner"
        style="width:120px;height:120px;font-size:24px"
      >
        <svg
          class="fishd-progress-circle "
          viewBox="0 0 100 100"
        >
          <path
            class="fishd-progress-circle-trail"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#f3f3f3"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
          />
          <path
            class="fishd-progress-circle-path"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#87d068"
            stroke-linecap="round"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
          />
        </svg>
        <div
          class="fishd-progress-circle-info"
        >
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-alone-success"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="fishd-progress fishd-progress-circle fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-inner"
        style="width:80px;height:80px;font-size:18px"
      >
        <svg
          class="fishd-progress-circle "
          viewBox="0 0 100 100"
        >
          <path
            class="fishd-progress-circle-trail"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#f3f3f3"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
          />
          <path
            class="fishd-progress-circle-path"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#108ee9"
            stroke-linecap="round"
            stroke-width="6"
            style="stroke-dasharray:88.59291283123217px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
          />
        </svg>
        <div
          class="fishd-progress-circle-info"
        >
          <span
            class="fishd-progress-text"
          >
            30%
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-circle fishd-progress-status-exception fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-inner"
        style="width:80px;height:80px;font-size:18px"
      >
        <svg
          class="fishd-progress-circle "
          viewBox="0 0 100 100"
        >
          <path
            class="fishd-progress-circle-trail"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#f3f3f3"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
          />
          <path
            class="fishd-progress-circle-path"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#ff5500"
            stroke-linecap="round"
            stroke-width="6"
            style="stroke-dasharray:206.7167966062084px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
          />
        </svg>
        <div
          class="fishd-progress-circle-info"
        >
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-alone-error"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-circle fishd-progress-status-success fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-inner"
        style="width:80px;height:80px;font-size:18px"
      >
        <svg
          class="fishd-progress-circle "
          viewBox="0 0 100 100"
        >
          <path
            class="fishd-progress-circle-trail"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#f3f3f3"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
          />
          <path
            class="fishd-progress-circle-path"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#87d068"
            stroke-linecap="round"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
          />
        </svg>
        <div
          class="fishd-progress-circle-info"
        >
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-alone-success"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="fishd-progress fishd-progress-circle fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-inner"
        style="width:120px;height:120px;font-size:24px"
      >
        <svg
          class="fishd-progress-circle "
          viewBox="0 0 100 100"
        >
          <path
            class="fishd-progress-circle-trail"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#f3f3f3"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
          />
          <path
            class="fishd-progress-circle-path"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#108ee9"
            stroke-linecap="round"
            stroke-width="0"
            style="stroke-dasharray:0px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
          />
        </svg>
        <div
          class="fishd-progress-circle-info"
        >
          <span
            class="fishd-progress-text"
          >
            0%
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-btn-group"
    >
      <button
        class="fishd-btn"
        type="button"
      >
        <span>
           - 
        </span>
      </button>
      <button
        class="fishd-btn"
        type="button"
      >
        <span>
           + 
        </span>
      </button>
    </div>
  </div>
  <div>
    <div
      class="fishd-progress fishd-progress-circle fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-inner"
        style="width:120px;height:120px;font-size:24px"
      >
        <svg
          class="fishd-progress-circle "
          viewBox="0 0 100 100"
        >
          <path
            class="fishd-progress-circle-trail"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#f3f3f3"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
          />
          <path
            class="fishd-progress-circle-path"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#108ee9"
            stroke-linecap="round"
            stroke-width="6"
            style="stroke-dasharray:221.48228207808043px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
          />
        </svg>
        <div
          class="fishd-progress-circle-info"
        >
          <span
            class="fishd-progress-text"
          >
            75 Days
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-circle fishd-progress-status-success fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-inner"
        style="width:120px;height:120px;font-size:24px"
      >
        <svg
          class="fishd-progress-circle "
          viewBox="0 0 100 100"
        >
          <path
            class="fishd-progress-circle-trail"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#f3f3f3"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
          />
          <path
            class="fishd-progress-circle-path"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#87d068"
            stroke-linecap="round"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
          />
        </svg>
        <div
          class="fishd-progress-circle-info"
        >
          <span
            class="fishd-progress-text"
          >
            Done
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:0%;height:8px;border-radius:100px"
              />
            </div>
          </div>
          <span
            class="fishd-progress-text"
          >
            0%
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-btn-group"
    >
      <button
        class="fishd-btn"
        type="button"
      >
        <span>
           - 
        </span>
      </button>
      <button
        class="fishd-btn"
        type="button"
      >
        <span>
           + 
        </span>
      </button>
    </div>
  </div>
  <div
    class="fishd-progress fishd-progress-circle fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
  >
    <div
      class="fishd-progress-inner"
      style="width:120px;height:120px;font-size:24px"
    >
      <svg
        class="fishd-progress-circle "
        viewBox="0 0 100 100"
      >
        <path
          class="fishd-progress-circle-trail"
          d="M 50,50 m 0,47
     a 47,47 0 1 1 0,-94
     a 47,47 0 1 1 0,94"
          fill-opacity="0"
          stroke="#f3f3f3"
          stroke-width="6"
          style="stroke-dasharray:220.30970943744057px 295.3097094374406px;stroke-dashoffset:-37.5px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
        />
        <path
          class="fishd-progress-circle-path"
          d="M 50,50 m 0,47
     a 47,47 0 1 1 0,-94
     a 47,47 0 1 1 0,94"
          fill-opacity="0"
          stroke="#108ee9"
          stroke-linecap="round"
          stroke-width="6"
          style="stroke-dasharray:165.23228207808043px 295.3097094374406px;stroke-dashoffset:-37.5px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
        />
      </svg>
      <div
        class="fishd-progress-circle-info"
      >
        <span
          class="fishd-progress-text"
        >
          75%
        </span>
      </div>
    </div>
  </div>
  <div
    class="fishd-progress fishd-progress-line fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
  >
    <div
      class="fishd-progress-line-ctner"
    >
      <div
        class="fishd-progress-basic"
      >
        <div
          class="fishd-progress-outer"
        >
          <div
            class="fishd-progress-inner"
          >
            <div
              class="fishd-progress-bg"
              style="width:60%;height:8px;border-radius:100px"
            />
            <div
              class="fishd-progress-success-bg"
              style="width:30%;height:8px;border-radius:100px"
            />
          </div>
        </div>
        <span
          class="fishd-progress-text"
        >
          60%
        </span>
      </div>
    </div>
  </div>
  <div
    class="fishd-progress fishd-progress-line fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
  >
    <div
      class="fishd-progress-line-ctner"
    >
      <div
        class="fishd-progress-basic"
      >
        <div
          class="fishd-progress-outer"
        >
          <div
            class="fishd-progress-inner"
          >
            <div
              class="fishd-progress-bg"
              style="width:65%;height:8px;border-radius:100px"
            >
              <div
                class="fishd-progress-extra"
              >
                <div
                  class="img-icon-14"
                />
              </div>
            </div>
          </div>
        </div>
        <span
          class="fishd-progress-text"
        >
          65%
        </span>
      </div>
    </div>
  </div>
  <div>
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-exception fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:70%;height:8px;border-radius:100px"
              />
            </div>
          </div>
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-error"
            />
          </span>
        </div>
        <span
          class="fishd-progress-msg"
        >
          网络错误
        </span>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-success fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:100%;height:8px;border-radius:100px"
              />
            </div>
          </div>
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-success"
            />
          </span>
        </div>
        <span
          class="fishd-progress-msg"
        >
          上传成功
        </span>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-circle fishd-progress-status-exception fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-inner"
        style="width:120px;height:120px;font-size:24px"
      >
        <svg
          class="fishd-progress-circle "
          viewBox="0 0 100 100"
        >
          <path
            class="fishd-progress-circle-trail"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#f3f3f3"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
          />
          <path
            class="fishd-progress-circle-path"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#ff5500"
            stroke-linecap="round"
            stroke-width="6"
            style="stroke-dasharray:206.7167966062084px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
          />
        </svg>
        <div
          class="fishd-progress-circle-info"
        >
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-alone-error"
            />
          </span>
          <span
            class="fishd-progress-msg"
          >
            网络错误
          </span>
        </div>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-circle fishd-progress-status-success fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-inner"
        style="width:120px;height:120px;font-size:24px"
      >
        <svg
          class="fishd-progress-circle "
          viewBox="0 0 100 100"
        >
          <path
            class="fishd-progress-circle-trail"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#f3f3f3"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
          />
          <path
            class="fishd-progress-circle-path"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#87d068"
            stroke-linecap="round"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
          />
        </svg>
        <div
          class="fishd-progress-circle-info"
        >
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-alone-success"
            />
          </span>
          <span
            class="fishd-progress-msg"
          >
            上传成功
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:30%;height:8px;border-radius:100px"
              />
            </div>
          </div>
          <span
            class="fishd-progress-text"
          >
            30%
          </span>
        </div>
        <span
          class="fishd-progress-oper"
        >
          <button
            class="fishd-btn fishd-progress-oper-btn fishd-btn-sm"
            type="button"
          >
            <span>
              取 消
            </span>
          </button>
        </span>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-line fishd-progress-status-exception fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-line-ctner"
      >
        <div
          class="fishd-progress-basic"
        >
          <div
            class="fishd-progress-outer"
          >
            <div
              class="fishd-progress-inner"
            >
              <div
                class="fishd-progress-bg"
                style="width:70%;height:8px;border-radius:100px"
              />
            </div>
          </div>
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-error"
            />
          </span>
        </div>
        <span
          class="fishd-progress-oper"
        >
          <button
            class="fishd-btn fishd-progress-oper-btn fishd-btn-sm"
            type="button"
          >
            <span>
              重 试
            </span>
          </button>
        </span>
      </div>
    </div>
    <div
      class="fishd-progress fishd-progress-circle fishd-progress-status-normal fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-inner"
        style="width:120px;height:120px;font-size:24px"
      >
        <svg
          class="fishd-progress-circle "
          viewBox="0 0 100 100"
        >
          <path
            class="fishd-progress-circle-trail"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#f3f3f3"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
          />
          <path
            class="fishd-progress-circle-path"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#108ee9"
            stroke-linecap="round"
            stroke-width="6"
            style="stroke-dasharray:88.59291283123217px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
          />
        </svg>
        <div
          class="fishd-progress-circle-info"
        >
          <span
            class="fishd-progress-text"
          >
            30%
          </span>
        </div>
      </div>
      <span
        class="fishd-progress-oper"
      >
        <button
          class="fishd-btn fishd-progress-oper-btn fishd-btn-sm"
          type="button"
        >
          <span>
            取 消
          </span>
        </button>
      </span>
    </div>
    <div
      class="fishd-progress fishd-progress-circle fishd-progress-status-exception fishd-progress-show-info fishd-progress-default"
    >
      <div
        class="fishd-progress-inner"
        style="width:120px;height:120px;font-size:24px"
      >
        <svg
          class="fishd-progress-circle "
          viewBox="0 0 100 100"
        >
          <path
            class="fishd-progress-circle-trail"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#f3f3f3"
            stroke-width="6"
            style="stroke-dasharray:295.3097094374406px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s"
          />
          <path
            class="fishd-progress-circle-path"
            d="M 50,50 m 0,-47
     a 47,47 0 1 1 0,94
     a 47,47 0 1 1 0,-94"
            fill-opacity="0"
            stroke="#ff5500"
            stroke-linecap="round"
            stroke-width="6"
            style="stroke-dasharray:206.7167966062084px 295.3097094374406px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s"
          />
        </svg>
        <div
          class="fishd-progress-circle-info"
        >
          <span
            class="fishd-progress-text"
          >
            <i
              class="fishdicon fishdicon-hints-alone-error"
            />
          </span>
        </div>
      </div>
      <span
        class="fishd-progress-oper"
      >
        <button
          class="fishd-btn fishd-progress-oper-btn fishd-btn-sm"
          type="button"
        >
          <span>
            重 试
          </span>
        </button>
      </span>
    </div>
  </div>
</div>
`;
